valhalla/sif/dynamiccost.h

```diff
@@ -433,7 +441,7 @@ class DynamicCost {
                                    const uint64_t current_time,
                                    const uint32_t tz_index,
                                    int& restriction_idx) const {
-    if (edge->access_restriction()) {
+    if (!ignore_restrictions_ && edge->access_restriction()) {
```

Comment:
@kevinkreiser we use `ignore_restrictions_` flag here, so it's not necessary to use it in `Allowed` methods

Reply:
this are complex restrictions, we still need to use it for simple restrictions

Reply:
`directededge::restrictions` == simple restrictions and `directededge::access_restriction` == complex restrictions

Reply:
as far as I can see, we handle simple restrictions earlier ( for example, here https://github.com/valhalla/valhalla/pull/2469/files#diff-645ef4e12203cf833688019b21401d07R427 ). do you want me to move simple restrictions processing from `Allowed` methods here ? or something else ?